app/controllers/admin/merchants_controller.rb

```diff
@@ -1,6 +1,26 @@
 class Admin::MerchantsController < ApplicationController
 
+  def index
+    @merchants = Merchant.all
+  end
+
   def show
     @merchant = Merchant.find(params[:merchant_id])
   end
+
+  def destroy
+    merchant = Merchant.find(params[:merchant_id])
+    merchant.update(status: 1)
+    merchant.deactivate_all_items
+    flash[:success] = "Merchant has been disabled"
+    redirect_to "/admin/merchants"
+  end
+
+  def update
+    merchant = Merchant.find(params[:merchant_id])
+    merchant.update(status: 0)
```

Comment:
Maybe worth a refactor later? see if we can build a helper method for destroy and update